File: obi2021f1/cifra.py
z = input()

def e_vogal(a: str) -> bool:
    return a == "a" or a == "e" or a == "i" or a == "o" or a == "u"

def vogal_mp(a: str) -> str:
    al = "abcdefghijklmnopqrstuvwxyz"
    p = 0
    an = 30
    dp = 30
    for i, x in enumerate(al):
        if x == a:
            p = i
            break

    for i in range(p, -1, -1):
        if e_vogal(al[i]):
            an = i
            break
    
    for i in range(p, len(al)):
        if e_vogal(al[i]):
            dp = i
            break

    if (p-an) <= dp - p:
        return al[an]
    else: 
        return al[dp]

r = ""

def cons_mp(a: str) -> str:
    al = "abcdefghijklmnopqrstuvwxyz"
    p = 00
    for i, x in enumerate(al):
        if x == a:
            p = i
            break

    for i in range(p+1, len(al)):
        if not e_vogal(al[i]):
            return al[i]

for i in z:
    if  e_vogal(i):
        r += i

    elif i == "z":
        r += i
        r += vogal_mp(i)
        r += i
    else:
        r += i
        r += vogal_mp(i)
        r += cons_mp(i)

print(r)

File: obi2021f1/cifra.cpp
#include <bits/stdc++.h>

#define ll long long int
#define nl endl

using namespace std;

bool e_vogal(char a){
    return a == 'a' or a == 'e' or a == 'i' or a == 'o' or a == 'u';
}

char vogal_mp(char c){
    string al = "abcdefghijklmnopqrstuvwxyz";

    int p = 0;

    for(int i = 0; i < al.size(); i++){
        if(al[i] == c){
            p = i;
            break;
        }
    }

    int a = 0, e = 4,i = 8,o = 14,u = 20;

    if(abs(p-a) < abs(p-e) and abs(p-a) < abs(p-i) and abs(p-a) < abs(p-o) and abs(p-a) < abs(p-u)) return 'a';
    else if(abs(p-e) < abs(p-a) and abs(p-e) < abs(p-i) and abs(p-e) < abs(p-o) and abs(p-e) < abs(p-u)) return 'e';
    else if(abs(p-i) < abs(p-a) and abs(p-i) < abs(p-e) and abs(p-i) < abs(p-o) and abs(p-i) < abs(p-u)) return 'i';
    else if(abs(p-o) < abs(p-a) and abs(p-o) < abs(p-e) and abs(p-o) < abs(p-i) and abs(p-o) < abs(p-u)) return 'o';
    else if(abs(p-u) < abs(p-a) and abs(p-u) < abs(p-e) and abs(p-u) < abs(p-i) and abs(p-u) < abs(p-o)) return 'u';
}

int main(void){
    string s, r;
    cin >> s;

    for(int i = 0; i < s.size(); i++){
        if (s[i] == 'z' or e_vogal(s[i])) s += s[i];
        else{
            r += s[i];
            r += vogal_mp(s[i]);
            r += s[i+1];
            i++;
        }
    }

    cout << r << nl;

    return 0;
} 